src/olympia/addons/tests/test_admin.py

```diff
@@ -726,3 +727,86 @@ def test_can_list_with_admin_curation_permission(self):
         response = self.client.get(self.list_url, follow=True)
         assert response.status_code == 200
         assert 'slugymcslugface' in response.content.decode('utf-8')
+
+
+class TestAddonRegionalRestrictionsAdmin(TestCase):
+    def setUp(self):
+        self.list_url = reverse(
+            'admin:addons_addonregionalrestrictions_changelist')
+        user = user_factory(email='someone@mozilla.com')
+        self.grant_permission(user, '*:*')
+        self.client.login(email=user.email)
+
+    def test_can_see_module_in_admin(self):
+        url = reverse('admin:index')
+        response = self.client.get(url)
+        assert response.status_code == 200
+
+        # Use django's reverse, since that's what the admin will use. Using our
+        # own would fail the assertion because of the locale that gets added.
+        self.list_url = django_reverse(
+            'admin:addons_addonregionalrestrictions_changelist')
+        assert self.list_url in response.content.decode('utf-8')
+
+    def test_can_list(self):
+        AddonRegionalRestrictions.objects.create(
+            addon=addon_factory(name='éléphant'), excluded_regions=['fr-FR'])
+        response = self.client.get(self.list_url, follow=True)
+        assert response.status_code == 200
+        assert b'fr-FR' in response.content
+        assert 'éléphant' in response.content.decode('utf-8')
+
+    def test_can_add(self):
+        addon = addon_factory()
+        self.add_url = reverse('admin:addons_addonregionalrestrictions_add')
+
+        response = self.client.get(self.add_url, follow=True)
+        assert response.status_code == 200
+        assert pq(response.content)('#id_addon')  # addon input is editable
+
+        response = self.client.post(
+            self.add_url, {
+                'excluded_regions': '["de", "pt-BR"]',
```

Comment:
Region code should be uppercase values only and I dont see that documented or enforced.

Reply:
I couldn't see that documented anywhere either - can you log an issue?

Reply:
I logged an issue https://github.com/mozilla/addons-server/issues/16023